lib/build_failure_notifier.rb

```diff
@@ -0,0 +1,98 @@
+require 'travis'
+require 'gitter/api'
+
+class BuildFailureNotifier
+  def self.gitter
+    @gitter ||= begin
+                  api_url = Settings.gitter.api_url
+                  client_settings = {
+                    :token      => Settings.gitter_credentials.token,
+                    :api_prefix => Settings.gitter.api_prefix,
+                    :api_uri    => api_url && URI(api_url)
+                  }
+
+                  Gitter::API::Client.new(client_settings)
+                end
+  end
+
+  def self.repo_room_map
+    settings = Settings.travis_branch_monitor
+    repo_map = settings.included_repos.to_h.merge(settings.excluded_repos.to_h)
+
+    repo_map.stringify_keys!
+    repo_map.each do |repo, room_uri|
+      repo_map[repo] = repo if room_uri.nil?
+    end
+  end
+
+  attr_reader :branch, :build, :repo, :repo_path, :room
+
+  def initialize(branch)
+    @branch    = branch
+    @repo_path = branch.repo.name
+    @repo      = Travis::Repository.find(repo_path)
+    @room      = repo_room_map[repo_path]
+    @build     = repo.session.find_one(Travis::Client::Build, branch.travis_build_failure_id)
+  end
+
+  def post_failure
+    notification_msg = <<~MSG
+      > ### :red_circle: Build Failure in #{repo_branches_markdown_url}!
+      >
+      > **Travis Build**:  #{travis_build_url}
+    MSG
+    notification_msg << "> **Failure PR**:    #{offending_pr}\n" if offending_pr
```

Comment:
It's not always a PR to that repo which causes the failure. Sometimes a merge to core breaks the API repo, but it is only noticed after the API repo cron runs.

Reply:
That is why the `if offending_pr` is there, which can return `nil`.

Reply:
However, I am open to suggestions on how we can improve this to better provide context for the message.  I realize that this could still display a false positive by displaying the "last merged commit" on something that was merged weeks ago, so maybe only display the PR if it was merged recently?  Within the last 3 days?